src/isomorphic/classic/types/ReactPropTypes.js

```diff
@@ -110,10 +111,21 @@ function createChainableTypeChecker(validate) {
     propName,
     componentName,
     location,
-    propFullName
+    propFullName,
+    secret
   ) {
     componentName = componentName || ANONYMOUS;
     propFullName = propFullName || propName;
+    if (!__DEV__) {
+      if (secret !== ReactPropTypesSecret && typeof console !== 'undefined') {
+        console.error(
```

Comment:
I would just log it once at all. Logging can get expensive in production (depends on the browser).